internal/controller/streams_helper.go

```diff
@@ -94,6 +94,9 @@ func streamMutationObjectData(sm *streamManager, metadata *pb.KubernetesObjectDa
 
 // sendClusterMetadata sends a message to indicate current cluster metadata
 func sendClusterMetadata(ctx context.Context, sm *streamManager) error {
+	// Wait until the flow collector decision is complete.
+	sm.streamClient.flowCollectorWG.Wait()
+
```

Comment:
I'm sorry, I gave you the wrong feedback.
We don't need synchronization / a WaitGroup. We're already doing things in the right order.
```suggestion
```